api/repositories/pod_repository.go

```diff
@@ -47,6 +47,54 @@ func (r *PodRepo) listPods(ctx context.Context, authInfo authorization.Info, lis
 	return podList.Items, nil
 }
 
+func (r *PodRepo) DeletePod(ctx context.Context, authInfo authorization.Info, appRevision string, appGUID string, appSpaceGUID string, instanceID string, processType string) error {
+	labelSelector, err := labels.ValidatedSelectorFromSet(map[string]string{
+		"korifi.cloudfoundry.org/app-guid":     appGUID,
+		"korifi.cloudfoundry.org/version":      appRevision,
+		"korifi.cloudfoundry.org/process-type": processType,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to build labelSelector: %w", apierrors.FromK8sError(err, PodResourceType))
+	}
+	listOpts := client.ListOptions{Namespace: appSpaceGUID, LabelSelector: labelSelector}
+
+	// find the pod name from the app
+	pods, err := r.listPods(ctx, authInfo, listOpts)
+	if err != nil {
+		return fmt.Errorf("failed to list pods: %w", apierrors.FromK8sError(err, PodResourceType))
+	}
+	if len(pods) == 0 {
+		return fmt.Errorf("no pods found for app and process: %w", apierrors.FromK8sError(err, PodResourceType))
+	}
+	pod := corev1.Pod{}
+	var found bool
+	for _, p := range pods {
+		if strings.HasSuffix(p.ObjectMeta.Name, instanceID) {
+			// delete the pod
+			pod = p
+			found = true
+			break
+		}
+	}
+	if !found {
+		return fmt.Errorf("instance not found: %w", apierrors.FromK8sError(err, PodResourceType))
+	}
+	if pod.Labels["korifi.cloudfoundry.org/process-type"] != processType {
```

Comment:
This cannot happen since the pods are listed with label selector that specifies the process type label

Reply:
ok, removed the additional check for the correct pod labels